### PR TITLE
Register login logout

### PR DESCRIPTION
**Pull request by Nothemba**

### CODE REVIEW

**Review by jens1101 — CHANGES_REQUESTED** [review]
The `node_modules` folder is still committed in spite of being in the gitignore file. Please run `git rm -r --cached node_modules` and commit in order to remove it. See [this SO answer](https://stackoverflow.com/a/62389965/1358008) for details.

**Comment by Nothemba** [comment]
Removed node modules, and also added a git ignore file on the backend
folder.

File: backend/routes/pages.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    
+    res.render('index');
+})
+
+
+router.get('/home', (req, res, next) => {
+    let user = req.session.user;
+
+    if(user) {
+        res.render('home', {opp:req.session.opp, name:user.username});
```
Comment (jens1101) [inline backend/routes/pages.js]: - Is a render necessary here? I believe you could just respond with JSON here.
- I cannot find any view named "home" in you code, so I don't believe this render will work at all.

**Review by jens1101 — CHANGES_REQUESTED** [review]
- `mysql` and `pug` are not used anywhere in this code. Why are they dependencies?
- When I run `npm start` I receive the following error:
  ```
  > moody-socials-backend@1.0.0 start /Users/jens/Documents/Projects/moody-socials/backend
  > cd server && nodemon server.js
  
  sh: line 0: cd: server: No such file or directory
  npm ERR! code ELIFECYCLE
  npm ERR! errno 1
  npm ERR! moody-socials-backend@1.0.0 start: `cd server && nodemon server.js`
  npm ERR! Exit status 1
  npm ERR! 
  npm ERR! Failed at the moody-socials-backend@1.0.0 start script.
  npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
  
  npm ERR! A complete log of this run can be found in:
  npm ERR!     /Users/jens/.npm/_logs/2021-01-27T11_47_30_747Z-debug.log
  
  Process finished with exit code 1
  ```

**Comment by Nothemba** [comment]
I've changed the startup script it works now, removed those dependencies
and unnecessary code.

**Review by jens1101 — CHANGES_REQUESTED** [review]
Any endpoint that I call responds with a "404 Page not found". Screenshot:

image

I believe this is because your error handling middleware in `server.js` is not correctly implemented.

Please test your changes thoroughly first before requesting another review.

**Review by jens1101 — APPROVED** [review]

**Pull request merged**